pkg/virt-dhcp/virt-dhcp.go

```diff
@@ -0,0 +1,105 @@
+package virtdhcp
+
+import (
+	"net"
+	"net/rpc"
+	"os"
+	"sync"
+
+	"kubevirt.io/kubevirt/pkg/log"
+)
+
+const DhcpSocket = "/var/run/kubevirt/virt_dhcp_sock"
+
+type DHCP struct {
+	name    string
+	dnsmasq *DNSmasqInstance
+	mutex   *sync.Mutex
+}
+
+type RemoveArgs struct {
+	IP  string
+	Mac string
+}
+
+type DhcpReply struct {
+	Message string
+	Success bool
+}
+
+type AddArgs struct {
+	IP    string
+	Mac   string
+	Lease int
+}
+
+func (s *DHCP) AddIP(args *AddArgs, reply *DhcpReply) error {
+	reply.Success = true
+	s.mutex.Lock()
+	err := s.dnsmasq.AddHost(args.Mac, args.IP, args.Lease)
+
+	if err != nil {
+		log.Log.Reason(err).Errorf("failed to add ip: %s and mac: %s to dhcp", args.IP, args.Mac)
+		reply.Success = false
+		reply.Message = err.Error()
+	}
+	s.mutex.Unlock()
+	return nil
+}
+
+func (s *DHCP) RemoveIP(args *RemoveArgs, reply *DhcpReply) error {
+	reply.Success = true
+	s.mutex.Lock()
+	err := s.dnsmasq.RemoveHost(args.IP)
+	if err != nil {
+		log.Log.Reason(err).Errorf("failed to remove ip: %s and mac: %s from dhcp", args.IP, args.Mac)
+		reply.Success = false
+		reply.Message = err.Error()
+	}
+	s.mutex.Unlock()
+	return nil
+}
+
+func (s *DHCP) SetKnownHosts(args *[]AddArgs, reply *DhcpReply) error {
+	reply.Success = true
+	s.mutex.Lock()
+	s.dnsmasq.SetKnownHosts(args)
+	s.mutex.Unlock()
+	return nil
+}
+
+func createSocket(socketPath string) (net.Listener, error) {
+
+	os.RemoveAll(socketPath)
+	socket, err := net.Listen("unix", socketPath)
+
+	if err != nil {
+		log.Log.Reason(err).Error("failed to create a socket for dhcp service")
+		return nil, err
+	}
+	return socket, nil
+}
+
+func Run() error {
+	var mutex = &sync.Mutex{}
+
+	rpcServer := rpc.NewServer()
+	dnsmasq := NewDNSmasq()
+	server := &DHCP{name: "virt-dhcp",
+		dnsmasq: dnsmasq,
+		mutex:   mutex}
+	rpcServer.Register(server)
+	dnsmasq.loadHostsFile()
```

Comment:
loadHostsFile returns an error. We should probably panic if that error is encountered. It would most likely be caused by a file permissions issue.